File: solutions/cpp/hamming/1/hamming.cpp
#include "hamming.h"

namespace hamming {
    int compute(const char* s1, const char* s2){
        int d=0;
        if (strlen(s1)!=strlen(s2)) throw std::domain_error("Unequal strands");
        for (int i=0;i<(int)strlen(s1);i++){
            if (s1[i]!=s2[i]) d++;
        }
        return d;
    }
}  // namespace hamming

File: solutions/fortran/hamming/1/hamming.f90
module hamming
  implicit none
contains

  function compute(strand1, strand2, distance)
      character(len=20) :: strand1
      character(len=20) ::  strand2
      character(len=20) ::  strand
      integer,intent(out) :: distance
      integer :: i, j
      logical :: compute
      strand=strand2
      if (len_trim(strand1) /= len_trim(strand)) then
        compute = .false.
        return
      end if
      distance = 0
      do i=1,len(strand1)
        if (strand1(i:i) == strand(i,i)) then
          distance = distance + 1
        end if
      end do
      return 

  end function compute

end module hamming
